From 656c73c0a7c22ae697585351805a748da616e05f Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:07 -0400
Subject: [usb] xhci: fix sparse warning in xhci-trace.h

Message-id: <1410887463-20674-176-git-send-email-dzickus@redhat.com>
Patchwork-id: 95308
O-Subject: [RHEL7 PATCH 175/591] xhci: fix sparse warning in xhci-trace.h
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d20c72256721411f55eb95beddf1d8a5d5da31eb
Author: Xenia Ragiadakou <burzalodowa@gmail.com>
Date:   Mon Sep 9 21:03:12 2013 +0300

    xhci: fix sparse warning in xhci-trace.h

    This patch fixes the following sparse warnings:
    drivers/usb/host/./xhci-trace.h:116:1: warning: cast to restricted __le32
    drivers/usb/host/./xhci-trace.h:116:1: warning: cast to restricted __le32
    drivers/usb/host/./xhci-trace.h:116:1: warning: restricted __le32 degrades to
    integer
    drivers/usb/host/./xhci-trace.h:116:1: warning: restricted __le32 degrades to
    integer

    by converting the field 'trb' of the trace buffer entry structure from array
    with elements of type __le32 to an array with elements of type u8.
    Into the trb array are copied the contents of the TRB that generated the event.
    The trace-cmd tool with the help of plugin_xhci.py will use this field to
    parse the TRB contents in a human readable way.

    Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-trace.h b/drivers/usb/host/xhci-trace.h
index afe0933..dde3959 100644
--- a/drivers/usb/host/xhci-trace.h
+++ b/drivers/usb/host/xhci-trace.h
@@ -116,7 +116,7 @@ DECLARE_EVENT_CLASS(xhci_log_event,
   __field(u64, dma)
   __field(u32, status)
   __field(u32, flags)
-  __dynamic_array(__le32, trb, 4)
+  __dynamic_array(u8, trb, sizeof(struct xhci_generic_trb))
  ),
  TP_fast_assign(
   __entry->va = trb_va;
-- 
1.7.1